### PR TITLE
SPEED-1: Add build system component

### PR DESCRIPTION
**Pull request by MitchellGray100**
### Description

This PR adds the repository building mechanism for inputs to the worker

### CODE REVIEW

File: workers/src/main/java/edu/cornell/repository/ConfigParser.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.List;
+
+/**
+ * Parses the config file of the repository.
+ */
+public class ConfigParser {
+    private EnumMap<Category, List<String>> configMap = new EnumMap<>(Category.class);
```
Comment (jakejack13) [inline workers/src/main/java/edu/cornell/repository/ConfigParser.java:14]: I would add some documentation to these fields (just for posterity sake)

File: workers/src/main/java/edu/cornell/repository/ConfigParser.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    private Path configFilePath;
+    public static class ConfigSyntaxException extends Exception {
```
Comment (jakejack13) [inline workers/src/main/java/edu/cornell/repository/ConfigParser.java:16]: I've been putting exceptions in their own files. Should we instead move them as static classes? Doesn't make a difference except in style

File: workers/src/main/java/edu/cornell/repository/ConfigParser.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.EnumMap;
+import java.util.List;
+
+/**
+ * Parses the config file of the repository.
+ */
+public class ConfigParser {
```
Comment (jakejack13) [inline workers/src/main/java/edu/cornell/repository/ConfigParser.java:13]: In the class comment, can you add documentation on what the syntax of the config file is? I'm struggling to understand how this one is laid out and parsed. Can you maybe add an example config file into this package as well for reference?

File: workers/src/main/java/edu/cornell/repository/ConfigParser.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    /**
+     * Gets the values associated with the config category.
+     * @param categoryType The type of category to get the information of.
+     * @return A List of the category values as strings or null if category doesn't exist.
+     */
+    public List<String> getConfigMapCategory(Category categoryType) {
+        return configMap.get(categoryType);
+    }
+
+    public enum Category {
```
Comment (jakejack13) [inline workers/src/main/java/edu/cornell/repository/ConfigParser.java:84]: Add documentation to this enum

**Review by jakejack13 — CHANGES_REQUESTED** [review]
Looks good overall but needs more documentation

File: workers/src/main/java/edu/cornell/repository/ConfigSyntaxException.java [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package edu.cornell.repository;
+
+public class ConfigSyntaxException extends Exception {
```
Comment (jakejack13) [inline workers/src/main/java/edu/cornell/repository/ConfigSyntaxException.java:3]: Documentation needed explaining what this exception is

File: workers/src/main/java/edu/cornell/repository/RepositoryBuildException.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package edu.cornell.repository;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+
+/**
+ * An exception that is generated when a repository build fails
+ */
+@EqualsAndHashCode(callSuper = true)
+@Data
```
Comment (jakejack13) [inline workers/src/main/java/edu/cornell/repository/RepositoryBuildException.java:10]: I'm dumb so please remove these two annotations

File: workers/src/main/java/edu/cornell/repository/Config.java [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+import java.nio.file.Path;
+import java.util.List;
+
+/**
+ * An object representing the SPEED config file in the repository
+ */
+public class Config {
+
+    private ConfigParser parser;
+    public Config(String configFilePath) throws ConfigSyntaxException, IOException {
```
Comment (jakejack13) [inline workers/src/main/java/edu/cornell/repository/Config.java:13]: Documentation needed on the field and the constructor

**Review by jakejack13 — CHANGES_REQUESTED** [review]
A couple more documentation things

**Review by owenxr — APPROVED** [review]
LGTM

File: workers/.idea/.gitignore
Comment (jakejack13) [inline workers/.idea/.gitignore:1]: Remove this file

File: workers/.idea/gradle.xml
Comment (jakejack13) [inline workers/.idea/gradle.xml:1]: Remove this file

File: workers/.idea/inspectionProfiles/Project_Default.xml
Comment (jakejack13) [inline workers/.idea/inspectionProfiles/Project_Default.xml:1]: Remove this file

File: workers/.idea/misc.xml
Comment (jakejack13) [inline workers/.idea/misc.xml:1]: Remove this file

File: workers/.idea/uiDesigner.xml
Comment (jakejack13) [inline workers/.idea/uiDesigner.xml:1]: Remove this file

File: workers/.idea/vcs.xml
Comment (jakejack13) [inline workers/.idea/vcs.xml:1]: Remove this file

**Review by jakejack13 — CHANGES_REQUESTED** [review]
Please remove the `workers/.idea` directory and add in the missing documentation that are from the remaining comments

**Review by jakejack13 — CHANGES_REQUESTED** [review]
Documentation good, just need those `workers/.idea` files deleted

**Review by jakejack13 — APPROVED** [review]
LGTM

**Pull request merged**